src/helpers.js

```diff
@@ -226,36 +261,38 @@ function updateField(field, requiredFields, node, formValues, logic, config) {
         }
       }
     });
+  };
 
   if (field.getComputedAttributes) {
-    const computedFieldValues = field.getComputedAttributes({
+    const newAttributes = field.getComputedAttributes({
       field,
       isRequired: fieldIsRequired,
       node,
       formValues,
       config,
       logic,
     });
-    updateValues(computedFieldValues);
+    updateAttributes(newAttributes);
   }
 
   // If field has a calculateConditionalProperties closure, run it and update the field properties
   if (field.calculateConditionalProperties) {
-    const newFieldValues = field.calculateConditionalProperties({
+    const { rootFieldAttrs, newAttributes } = field.calculateConditionalProperties({
       isRequired: fieldIsRequired,
       conditionBranch: node,
       formValues,
     });
-    updateValues(newFieldValues);
+    updateAttributes(newAttributes, rootFieldAttrs);
+    removeConditionalStaleAttributes(field, newAttributes, rootFieldAttrs);
```

Comment:
note: In the first commit the `removeConditionalStaleAttributes` was inside `updateAttributes` so it would be called for every dynamic scenario (conditionals, custom validations, and computed attrs). But tha it caused other tests to fail, and I couldn't understand why in a short time.

As this bug is specifically focused on "leaked" attributes in conditionals, I decided to focus the solution only on that for now. I never saw bugs reported from the other dynamic scenarios, so... as they say "[if it ain't broke, don't fix it](https://en.wikipedia.org/wiki/wikt:if_it_ain%27t_broke,_don%27t_fix_it)." 

A conscious trade-off between between speed and excellence.